packages/db/index.js

```diff
@@ -139,6 +139,28 @@ export class DBClient {
     return data.length ? data[0] : undefined
   }
 
+  /**
+   * Check that a user account is restricted.
+   *
+   * @param {number} userId
+   * @returns {Promise<boolean>}
+   */
+  async isAccountRestricted (userId) {
+    const { error, count } = await this._client
+      .from('user_tag')
+      .select('id', { count: 'exact' })
+      .eq('user_id', userId)
+      .eq('tag', 'HasAccountRestriction')
+      .eq('value', true)
+      .filter('deleted_at', 'is', null)
+
+    if (error) {
+      throw new DBError(error)
+    }
+
```

Comment:
Expecting only one entry for this and pinning auth.

Reply:
I might be misunderstanding this, but why would this query becomes invalid if the `user_tag_is_not_deleted_idx` index is removed? Wouldn't that just make the query slower? It wouldn't make it not work, right?

Reply:
If that index is removed multiple `deleted` tags would be allowed isn't it? we'd remove the unique constraint. 
In that scenario, just counting wouldn't be enough. Wouldn't it? Because I would count an "out-of-date" tag.

In that scenario the truth would come from the latest tag.
Hence I was suggesting ways to prevent that if that constraint ever gets removed.
Is it making sense?


We should add a ticket for psa access as well!

Reply:
Ah, right, yes, got you. I think we could fix this (for all user tags) by implementing my suggestion here: https://github.com/web3-storage/web3.storage/pull/1053#discussion_r825887574